clouditor-engine-core/src/main/java/io/clouditor/data_access_layer/HibernateUtils.java

```diff
@@ -0,0 +1,171 @@
+          .addAnnotatedClass(FilteredAssetType.class)
+          .addAnnotatedClass(Asset.class)
+          .addAnnotatedClass(Condition.class)
+          .addAnnotatedClass(Scan.class)
+          .addAnnotatedClass(DiscoveryResult.class)
+          .addAnnotatedClass(User.class)
+          .setProperty("hibernate.enable_lazy_load_no_trans", "true")
+          .setProperty("hibernate.dialect", "org.hibernate.dialect.PostgreSQL94Dialect")
+          // Enable an automatic generation of the data model.
+          .setProperty("hibernate.hbm2ddl.auto", "create");
+
+  private static SessionFactory sessionFactory;
+  private static Session session;
+
+  /**
+   * Getter for the current session. If there is no current session, it opens a new session.
+   *
+   * @return the current session
+   * @throws IllegalStateException if the database was not initialized.
+   */
+  public static Session getSession() {
+    if (session == null) setSession(getSessionFactory().openSession());
+    return session;
+  }
+
+  /**
+   * Connects Hibernate to the persistent PostgreSQL database at the address:
+   * "jdbc:postgresql://{host}:{port}/{dbName}".
+   *
+   * @param host the host of the database.
+   * @param port the port of the database should be in range [1024; 49151].
+   * @param dbName the database name.
+   * @param userName the user name.
+   * @param password the password of the user
+   * @throws IllegalArgumentException if the <code>port</code> is not in range [1024; 49151].
+   * @throws NullPointerException if <code>host</code>, <code>dbName</code>, <code>userName</code>
+   *     or <code>password</code> is null.
+   */
+  public static void init(
+      final String host,
+      final int port,
+      final String dbName,
+      final String userName,
+      final String password) {
+    Objects.requireNonNull(host);
+    Objects.requireNonNull(dbName);
+    Objects.requireNonNull(userName);
+    Objects.requireNonNull(password);
+    if (port < 1024 || port > 49151)
+      throw new IllegalArgumentException(
+          "The given port: " + port + ", was not in range [1024; 49151].");
+    CONFIGURATION
+        .setProperty("hibernate.connection.driver_class", "org.postgresql.Driver")
+        .setProperty(
+            "hibernate.connection.url", "jdbc:postgresql://" + host + ":" + port + "/" + dbName);
+    setUserNameAndPassword(userName, password);
+    buildSessionFactory();
+  }
+
+  /**
+   * Connects to a automatically created H2 in memory database at the address: "jdbc:h2:~/{dbName}".
+   *
+   * @param dbName the database name.
+   * @param userName the user name.
+   * @param password the password of the user
+   * @throws IllegalArgumentException if the <code>port</code> is not in range [1024; 49151].
+   * @throws NullPointerException if <code>dbName</code>, <code>userName</code> or <code>password
+   *     </code> is null.
+   */
+  public static void init(final String dbName, final String userName, final String password) {
```

Comment:
Please rename this to something like initH2. there is no easy way to distinguish this from the other `init` function